sonar-orchestrator/src/main/java/com/sonar/orchestrator/build/dotnet/scanner/PackageDetailsFactory.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Orchestrator
+ * Copyright (C) 2011-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package com.sonar.orchestrator.build.dotnet.scanner;
+
+import com.sonar.orchestrator.version.Version;
+
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nonnull;
+
+public class PackageDetailsFactory {
+  // Starting with this version the download paths have changed.
+  public static final Version PACKAGING_CHANGE_VERSION = Version.create("6.0.0.81631");
+
+  public PackageDetails create(@NotNull Version scannerVersion, boolean useDotNetCore) {
+    // Depending on the version of the scanner, the file names, the artifactId and the classifiers can be different.
+    // Before 6.0.0.81631, the file names had the following format:
```

Comment:
NP: This explanation is valuable, but then when I read what `hasNewPackageNames()` does, I was confused. After looking at _sonar-scanner-msbuild_ releases, I understood that by _"Before 6.0.0.81631"_ you meant before the 6 major version entirely (so 5 and below). I would turn it this to make it clearer, like "For versions 5.x and below ... Starting version 6.x ..."